### DISCUSSION TITLE
Always show Bad Gateway error when use Foxglove in ROS1 environment

### DISCUSSION THREAD

**Discussion by LeZhang13** [foxglove/community#60] [Q&A] [answered]
Description
  Version:1.13.2
  OS:Ubantu 18.04 (Virtual Machine）
  Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native):ROS 1 native
Problem:
My ROS publisher and Foxglove all in my Virtual Machine Ubantu 18.04, I use a publisher publish the /Camea/Image, and want use Foxglove to show the web camera image by TCP/IP , but Foxglove always show error: Bad Gateway.
![image](https://user-images.githubusercontent.com/89899260/173288533-36e5525c-c8b7-431e-b012-4db5012fe256.png)
![image](https://user-images.githubusercontent.com/89899260/173288591-345961cd-37f5-4733-9b0f-1ccb80b20281.png)
![image](https://user-images.githubusercontent.com/89899260/173288646-f9c803ec-67d2-4fcd-a78f-0bf2318befd9.png)
![image](https://user-images.githubusercontent.com/89899260/173288718-2768944a-47da-4396-b14b-86c94f2db62d.png)
I don't know what problem caused.
By the way, If  I change to use rosbridge, the connection is ok, Foxglove can show the web camera image.

**Comment by jtbandes** [comment]
We do recommend the rosbridge connection since it's easier to get it to work (as you have found). It also works with the web app at studio.foxglove.dev and not just the desktop app.

The error message you shared mentions a hostname of `tedzhang-virtual-machine`. While this doesn't show up in your `printenv`, this error does indicate that at least one node is reporting this hostname, so it might be in the environment (`ROS_HOSTNAME`) for _some_ of your nodes (or roscore) running in different shells. It is important to ensure that each node in the system uses a hostname that can be resolved by other nodes it needs to communicate with.

Here is some ROS documentation about network setup: https://wiki.ros.org/ROS/NetworkSetup

**Reply by LeZhang13** [reply]
Hi, I export the HOSTNAME same as the error message show tedzhang-virtual-machine 
![image](https://user-images.githubusercontent.com/89899260/173479033-c4bf1ad7-9adc-43cf-a5bf-914688471988.png)
But after I connect Foxglove, same error happened.
![image](https://user-images.githubusercontent.com/89899260/173479132-fe3f2d3a-16e5-4280-a94e-52dd8ec9550e.png)
![image](https://user-images.githubusercontent.com/89899260/173479181-fdb16b65-1ee7-4bde-8aca-4e6df17ae0b2.png)
So I think this may not HOSTNAME problem. I have no idea to use native connection now.

**Comment by LeZhang13** [comment, marked as answer]
I change to use a local environment like win10, this issue fixed. So maybe caused by virtual environment issue.
